File: OpenAIJsonWrapper/node/examples/text_analysis.js
import OpenAI from "openai";
import { OpenAIJsonWrapper } from "../src/index.js";

const apiKey = process.env.OPENAI_API_KEY || "your-api-key-here";
const baseUrl = process.env.OPENAI_BASE_URL || "https://api.minimaxi.com/v1";
const modelName = process.env.OPENAI_MODEL_NAME || "MiniMax-M3";

if (apiKey === "your-api-key-here") {
  console.log("警告: 未检测到 OPENAI_API_KEY 环境变量，请确保已设置或手动修改脚本逻辑。");
}

const client = new OpenAI({
  apiKey,
  baseURL: baseUrl
});

const targetStructure = {
  analysis: {
    sentiment: "string (Positive/Negative/Neutral)",
    key_entities: ["string"],
    confidence_score: "float (0-1)"
  },
  response_suggestion: "string"
};

const wrapper = new OpenAIJsonWrapper(client, {
  model: modelName,
  targetStructure
});

const messages = [
  { role: "user", content: "GitHub Copilot 是一款非常棒的 AI 编程助手，它能极大地提高代码生产力，虽然偶尔会有小瑕疵，但整体瑕不掩瑜。" }
];

const requirements = [
  "情感分析必须细化到分值",
  "key_entities 至少提取两个",
  "response_suggestion 必须是中文"
];

const background = "你是一个专业级的产品评论分析专家，擅长从用户反馈中提取结构化洞察。";

async function main() {
  console.log("--- 正在发送请求到 LLM ---");

  const result = await wrapper.chat(messages, {
    requirements,
    background
  });

  console.log("\n--- 解析结果 ---");
  if (!result.error) {
    console.log("成功解析数据:");
    console.log(JSON.stringify(result.data, null, 2));
    console.log("\n思维链/推理过程:");
    console.log(result.reasoning);
  } else {
    console.log("解析失败!");
    console.log("错误信息:", result.error);
    console.log("原始响应内容:\n", result.raw_content);
  }
}

main();

File: OpenAIJsonWrapper/rust/examples/text_analysis.rs
use openai_json_wrapper::{OpenAIClientBuilder, OpenAIJsonWrapper, Message, MessageContent, ChatOptions};
use serde_json::json;
use std::env;

fn main() {
    for path in [".env", "../.env", "../../.env"] {
        if dotenvy::from_filename(path).is_ok() {
            break;
        }
    }

    let api_key = env::var("OPENAI_API_KEY").unwrap_or_else(|_| {
        println!("警告: 未检测到 OPENAI_API_KEY，请检查 .env 文件或设置环境变量");
        "your-api-key-here".to_string()
    });
    let base_url = env::var("OPENAI_BASE_URL").unwrap_or_else(|_| "https://api.minimaxi.com/v1".to_string());
    let model_name = env::var("OPENAI_MODEL_NAME").unwrap_or_else(|_| "MiniMax-M3".to_string());

    if api_key == "your-api-key-here" {
        println!("警告: 未检测到 OPENAI_API_KEY 环境变量，请确保已设置或手动修改脚本逻辑。");
    }

    let target_structure = json!({
        "analysis": {
            "sentiment": "string (Positive/Negative/Neutral)",
            "key_entities": ["string"],
            "confidence_score": "float (0-1)"
        },
        "response_suggestion": "string"
    });

    let client = OpenAIClientBuilder::new(&api_key)
        .base_url(&base_url)
        .build();

    let wrapper = OpenAIJsonWrapper::new(
        Box::new(client),
        &model_name,
        Some(target_structure.clone()),
        None,
        Some("You are a professional product review analyst."),
    );

    let messages = vec![Message {
        role: "user".to_string(),
        content: MessageContent::String("GitHub Copilot 是一款非常棒的 AI 编程助手，它能极大地提高代码生产力，虽然偶尔会有小瑕疵，但整体瑕不掩瑜。".to_string()),
    }];

    let options = ChatOptions {
        requirements: Some("情感分析必须细化到分值".to_string()),
        extra_requirements: Some("response_suggestion 必须是中文".to_string()),
        background: Some("你是一个专业级的产品评论分析专家，擅长从用户反馈中提取结构化洞察。".to_string()),
        ..Default::default()
    };

    println!("--- 发送请求 ---\n");

    let result = wrapper.chat(messages, options).expect("chat failed");

    println!("\n--- 解析结果 ---");
    if let Some(error) = result.error {
        println!("解析失败!");
        println!("错误信息: {}", error);
        println!("原始响应:\n{}", result.raw_content);
    } else {
        println!("成功解析数据:");
        println!("{}", serde_json::to_string_pretty(&result.data).unwrap());
        println!("\n思维链/推理过程:");
        println!("{}", result.reasoning);
    }
}
